### PR TITLE
Add access token scope checks to improve authorization

### PR DESCRIPTION
**Pull request by davidallendj**
This PR adds the ability for SMD to check for scopes and implements two related functions: `VerifyClaims` and `VerifyScope`. It also moves dedicated OAuth functionality to a separate, dedicated file in `cmd/smd/auth.go`.

The `VerifyClaims` function is needed to make sure that the `scope` claim is included in the incoming access token. It can also be used later for checking for inclusion of other claims such as `partition-id` required for accessing specific resources.

The `VerifyScope` function is used to check the incoming access token for specific scopes for each endpoint. Currently, this has only been implemented for `/State/Components` and `/Inventory/RedfishEndpoints` with each requiring the `smd_read` scope. Submitting a valid token without the `smd_read` scope will return the following error:

```bash
curl http://127.0.0.1:27779/hsm/v2/Inventory/RedfishEndpoints -H "Authorization: Bearer eyJhbGciOi..."
{"type":"about:blank","title":"Unauthorized","detail":"failed to verify scope: invalid or missing scope","status":401}
```
However, for any other endpoint that is not checking for scopes, any valid access token is sufficient to use like before.

### CODE REVIEW

**Review by synackd — CHANGES_REQUESTED** [review]
Looks good and tests are passing.

**Comment by synackd** [comment]
Once https://github.com/go-chi/jwtauth/pull/85 is merged, [06b7790](https://github.com/OpenCHAMI/smd/pull/17/commits/06b779026944519d3b8776d1ce1ef12d64d1fc26) should be reverted.

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,8 @@ go 1.21
 
 toolchain go1.21.5
 
+replace github.com/go-chi/jwtauth/v5 => ../jwtauth
```
[the remark this answers is not in the document]
Comment (davidallendj) [reply go.mod]: FYI I went ahead and made another fork so I could change the module as well. The other fork is the one that supposed to *eventually* get merged back upstream.

**Comment by synackd** [comment]
Runner tests passing.

**Review by synackd — APPROVED** [review]

**Pull request merged**